### PR TITLE
Added im-services-analytics consumer

### PR DESCRIPTION
**Pull request by varunreddy-egov**

### CODE REVIEW

File: backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/service/UpdateService.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+						log.error("ES is DOWN, Pausing kafka listener.......");
+						indexerUtils.orchestrateListenerOnESHealth();
+					} catch (Exception e) {
+						log.error("Exception while trying to update the ES document. Note: ES is not Down.", e);
+					}
+				}
+			}
+		} catch (Exception e) {
+			log.error("Error while building curl for updating", e);
+		}
```
[the remark this answers is not in the document]
Comment (subhashini-egov) [reply backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/service/UpdateService.java]: @varunreddy-egov can you check this coderabbit review comment and make appropriate changes?

File: backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/kafka/consumer/EventConsumerConfig.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+			kafkContainer = container;
+		} catch (Exception e) {
+			log.error("Container couldn't be started: ", e);
+			return false;
+		}
+		kafkContainer.start();
+		log.info("Custom KakfaListenerContainer STARTED...");
+		return true;
+
+	}
```
[the remark this answers is not in the document]
Comment (subhashini-egov) [reply backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/kafka/consumer/EventConsumerConfig.java:146]: @varunreddy-egov this as well.

File: backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/util/UpdateUtils.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+		// JsonPath cannot be applied on the type JSONObject. String has to be built of
+		// it and then used.
+		String[] array = object.toString().split(":");
+		StringBuilder jsonArray = new StringBuilder();
+		for (int i = 0; i < array.length; i++) {
+			jsonArray.append(array[i]);
+			if (i != array.length - 1)
+				jsonArray.append(":");
+		}
+		return jsonArray.toString();
```
[the remark this answers is not in the document]
Comment (subhashini-egov) [reply backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/util/UpdateUtils.java:98]: Let's rename the method. @varunreddy-egov

File: backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/util/UpdateUtils.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+		String[] array = jsonString.split(":");
+		StringBuilder jsonArray = new StringBuilder();
+		for (int i = 1; i < array.length; i++) {
+			jsonArray.append(array[i]);
+			if (i != array.length - 1)
+				jsonArray.append(":");
+		}
+		jsonArray.deleteCharAt(jsonArray.length() - 1);
+
+		return jsonArray.toString();
```
[the remark this answers is not in the document]
Comment (subhashini-egov) [reply backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/util/UpdateUtils.java:117]: @varunreddy-egov see if this makes sense. Why are we not using JSONObject to parse the string?

**Comment by subhashini-egov** [comment]
@varunreddy-egov please resolve the code suggestions and then notify for merge.

File: backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/config/ConsumerConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	private String updateIndexPath;
+
+	@Value("${elasticsearch.poll.interval.seconds}")
+	private String pollInterval;
+
+	@Value("${egov.indexer.es.username}")
+	private String esUsername;
+
+	@Value("${egov.indexer.es.password}")
+	private String esPassword;
```
[the remark this answers is not in the document]
Comment (varunreddy-egov) [reply backend/core-services/im-services-analytics/src/main/java/org/selco/e4h/config/ConsumerConfiguration.java:40]: Password is taken from helm charts not from properties

**Review by subhashini-egov — APPROVED** [review]

**Pull request merged**